api/prometheus/v1/api_bench_test.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2017 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package v1
+
+import (
+	"encoding/json"
+	"strconv"
+	"testing"
+	"time"
+
+	jsoniter "github.com/json-iterator/go"
+	"github.com/prometheus/common/model"
+)
+
+func generateData(timeseries, datapoints int) model.Matrix {
+	// Create the top-level matrix
+	m := make(model.Matrix, 0)
+
+	for i := 0; i < timeseries; i++ {
+		lset := map[model.LabelName]model.LabelValue{
+			model.MetricNameLabel: model.LabelValue("timeseries_" + strconv.Itoa(i)),
+		}
+		now := model.Now()
+		values := make([]model.SamplePair, datapoints)
+
+		for x := datapoints; x > 0; x-- {
+			values[x-1] = model.SamplePair{
+				Timestamp: now.Add(time.Second * -15 * time.Duration(x)), // Set the time back assuming a 15s interval
```

Comment:
I used 14s at the time because that was my scrape interval, for benchmarking it only matters how many points we are marshaling (the spacing doesn't actually matter)

Reply:
Aah ok, Thanks. 
In that case let's just increment it by 1s or something similar  to avoid any confusion that it might actually have some special meaning.